app/src/main/java/com/duckduckgo/app/generalsettings/GeneralSettingsViewModel.kt

```diff
@@ -0,0 +1,152 @@
+import com.duckduckgo.voice.store.VoiceSearchRepository
+import javax.inject.Inject
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+@ContributesViewModel(ActivityScope::class)
+class GeneralSettingsViewModel @Inject constructor(
+    private val settingsDataStore: SettingsDataStore,
+    private val pixel: Pixel,
+    private val history: NavigationHistory,
+    private val voiceSearchAvailability: VoiceSearchAvailability,
+    private val voiceSearchRepository: VoiceSearchRepository,
+    private val dispatcherProvider: DispatcherProvider,
+) : ViewModel() {
+
+    data class ViewState(
+        val autoCompleteSuggestionsEnabled: Boolean,
+        val autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled: Boolean,
+        val storeHistoryEnabled: Boolean,
+        val showVoiceSearch: Boolean,
+        val voiceSearchEnabled: Boolean,
+    )
+
+    private val viewState = MutableStateFlow(
+        ViewState(
+            autoCompleteSuggestionsEnabled = settingsDataStore.autoCompleteSuggestionsEnabled,
+            autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled = runBlocking { history.isHistoryUserEnabled() },
+            storeHistoryEnabled = history.isHistoryFeatureAvailable(),
+            showVoiceSearch = voiceSearchAvailability.isVoiceSearchSupported,
+            voiceSearchEnabled = voiceSearchAvailability.isVoiceSearchAvailable,
+        ),
+    )
+
+    fun viewState(): Flow<ViewState> = viewState.onStart {
+        viewModelScope.launch(dispatcherProvider.io()) {
+            val autoCompleteEnabled = settingsDataStore.autoCompleteSuggestionsEnabled
+            if (!autoCompleteEnabled) {
+                history.setHistoryUserEnabled(false)
+            }
+            withContext(dispatcherProvider.main()) {
+                viewState.emit(
+                    ViewState(
+                        autoCompleteSuggestionsEnabled = settingsDataStore.autoCompleteSuggestionsEnabled,
+                        autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled = history.isHistoryUserEnabled(),
+                        storeHistoryEnabled = history.isHistoryFeatureAvailable(),
+                        showVoiceSearch = voiceSearchAvailability.isVoiceSearchSupported,
+                        voiceSearchEnabled = voiceSearchAvailability.isVoiceSearchAvailable,
+                    ),
+                )
+            }
+        }
+    }
+
+    fun onAutocompleteSettingChanged(enabled: Boolean) {
+        Timber.i("User changed autocomplete setting, is now enabled: $enabled")
+        viewModelScope.launch(dispatcherProvider.io()) {
+            settingsDataStore.autoCompleteSuggestionsEnabled = enabled
+            if (!enabled) {
+                viewModelScope.launch() {
```

Comment:
this one seems not necessary since we are already inside a coroutine and io

Reply:
Again, this is copied from `PrivateSearchViewModel`

Reply:
Removed it from here and in `PrivateSearchViewModel`